auraxium/event/_trigger.py

```diff
@@ -212,10 +212,14 @@ def generate_subscription(self, logical_and: Optional[bool] = None) -> str:
         # When subscribing to character-centric events using only a world ID,
         # set the "logicalAnd*" flag to avoid subscribing to all characters on
         # all continents (characters would default to "all" if not specified).
-        elif (self.worlds and not self.characters
-                and any((issubclass(e, CharacterEvent))  # type: ignore
-                        for e in self.events)):
-            json_data['logicalAndCharactersWithWorlds'] = 'true'
+        elif (self.worlds and not self.characters):
+            charEventClsNames = [subclass.__name__ for subclass in CharacterEvent.__subclasses__()]
+            charEventStrings = [event for event in self.events if isinstance(event, str) and event in charEventClsNames]
+            if charEventStrings or any( isinstance(event, (CharacterEvent)) for event in self.events ):
+                json_data['logicalAndCharactersWithWorlds'] = 'true'
+            # else:
+            #     json_data['logicalAndCharactersWithWorlds'] = 'false'
+
```

Comment:
This dead code can go - we intentionally do not (un)set the logicalAnd for any other case as there may be new events or side effects of having it set vs. defaulted. We auto-insert it for character events with world filters as the worlds filter cannot work otherwise; it would just return all events for all worlds.

Reply:
Ah, I see.  Wasn't too sure about that so just had it there just in case!